### ISSUE TITLE
MAP can't update this Uniform Variable

### ISSUE THREAD

**Issue by cavities** [blei-lab/edward#733]
i use ed.MAP  a variable but this variable is Uniform,when  update inference every time,this uniform is not updated; this var value is   stochastic value in queue, and use PointMass estimate this point;

**Comment by dustinvtran** [comment]
Can you provide a minimal example?

**Comment by cavities** [comment]
this is an sample use MAP pointMass this tau but ..
`import edward as ed
import tensorflow as tf
from edward.models import Exponential,Uniform,Poisson,Empirical,PointMass,Normal,Laplace
import numpy as np

count_data = np.array([13,  24,   8,  24,   7,  35,  14,  11,  15,  11,  22,  22,  11,  57,  11,
  19,  29,   6,  19,  12,  22,  12,  18,  72,  32,   9,   7,  13,  19,  23,
  27,  20,   6,  17,  13,  10,  14,   6,  16,  15,   7,   2,  15,  15,  19,
  70,  49,   7,  53,  22,  21,  31,  19,  11,  18,  20,  12,  35,  17,  23,
  17,   4,   2,  31,  30,  13,  27,   0,  39,  37,   5,  14,  13,  22])

n_count_data = len(count_data)

alpha_f = 1.0/count_data.mean()
alpha = tf.Variable(alpha_f, name="alpha", dtype=tf.float32)
prior_lambda_1 = Exponential(float(alpha_f))
prior_lambda_2 = Exponential(float(alpha_f))
tau = Uniform(low=0.0,high=float(n_count_data - 1))
idx = np.arange(n_count_data)
lambda_ = tf.where(tau>=idx,tf.ones(shape=(n_count_data),dtype=tf.float32)*prior_lambda_1,tf.ones(shape=(n_count_data),dtype=tf.float32)*prior_lambda_2)
edpoi = Poisson(rate=lambda_)

post_lambda_1 = PointMass(params = tf.Variable(1.0))
post_lambda_2 = PointMass(params = tf.Variable(1.0))
post_tau = PointMass(params =tf.Variable(1.0))
lv = {
    prior_lambda_1:post_lambda_1,prior_lambda_2:post_lambda_2,tau:post_tau
}
inference = ed.MAP(lv,{edpoi:count_data})
inference.run()

for x in range(1000):
    info_dict = inference.update()
    tau_update = post_tau.eval()
    print(tau_update)`

**Comment by dustinvtran** [comment]
A boolean is a piecewise constant so gradient descent with MAP will fail for optimizing the parameters of a condition statement. See the question and answer in https://stackoverflow.com/questions/43544007/tf-where-causes-optimiser-to-fail-in-tensorflow/43545088.
